### ISSUE TITLE
Format `lock.json` and `deno test --coverage` files with deno fmt

### ISSUE THREAD

**Issue by starkpsmichael** [denoland/deno#9693]
Now that `deno fmt` can format json(c) files, all generated `json` files from deno, like `lock.json` and `deno test --coverage` files, should be formatted with it, before they get written.

### Actual Result `lock.json`
```
{
{
```

### Actual Result `deno test --coverage`
```
{"scriptId":"36","url":"deno:runtime/js/12_io.js","functions":[{"functionName":"writeSync","ranges":[{"startOffset":2185,"endOffset":2358,"count":4},{"startOffset":2289,"endOffset":2334,"count":0}],"isBlockCoverage":true}]}
```

### Expected Result `lock.json`
```
{
{
// notice the newline
```

### Expected Result `deno test --coverage`
```
{
  "scriptId": "36",
  "url": "deno:runtime/js/12_io.js",
  "functions": [
    {
      "functionName": "writeSync",
      "ranges": [
        { "startOffset": 2185, "endOffset": 2358, "count": 4 },
        { "startOffset": 2289, "endOffset": 2334, "count": 0 }
      ],
      "isBlockCoverage": true
    }
  ]
}
// notice the newline
```

**Comment by bartlomieju** [comment]
This is very reasonable request and would be quite easy to implement. I'm in favor of doing that and contributions are welcome!

**Comment by WenheLI** [comment]
Hi, interested in this issue; Just wondering what would be the possible starting point?

**Comment by bartlomieju** [comment]
Hi @WenheLI, thanks for interest working on this. These steps should get you off the ground:

- there's `format_json` function available in `cli/tools/fmt.rs` that you can use to format JSON output:
https://github.com/denoland/deno/blob/d31378726e78490e88b8a9ec3001b86ea009d978/cli/tools/fmt.rs#L209-L218

- for lockfile, you should run the output through `format_json` before saving it to file here:
https://github.com/denoland/deno/blob/d31378726e78490e88b8a9ec3001b86ea009d978/cli/lockfile.rs#L44

- for coverage it's slightly more involved as you need to first rewrite function to write all reports to a buffer and only then format them and write to file, around here:
https://github.com/denoland/deno/blob/d31378726e78490e88b8a9ec3001b86ea009d978/cli/tools/coverage.rs#L156-L164

Let me know if you have any more questions.

**Comment by WenheLI** [comment]
@bartlomieju - Hi, I am implementing the core logic here. However it seems that the deno does not contain test cases for checking lockfile. Should we also add some test cases here?

**Comment by bartlomieju** [comment]
Fixed in #13018
